[Kura 3.1.0 QA] SensorTagDriver NPE if wrong address

SensorTagDriver throws NPE that propagates to REST service in case channel's `sensortag.address` is incorrect:

```
Caused by: java.lang.NullPointerException
        at org.eclipse.kura.internal.driver.ble.sensortag.SensorTagDriver.getSensorTag(SensorTagDriver.java:330)
        at org.eclipse.kura.internal.driver.ble.sensortag.SensorTagDriver.runReadRequest(SensorTagDriver.java:258)
        at org.eclipse.kura.internal.driver.ble.sensortag.SensorTagDriver.access$2(SensorTagDriver.java:257)
        at org.eclipse.kura.internal.driver.ble.sensortag.SensorTagDriver$SensorTagPreparedRead.execute(SensorTagDriver.java:474)
        at org.eclipse.kura.asset.provider.BaseAsset.readAllChannels(BaseAsset.java:492)
        at org.eclipse.kura.internal.rest.asset.AssetRestService.read(AssetRestService.java:85)
```

So the result is a rather uninformative
```
<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 500 </title>
</head>
<body>
<h2>HTTP ERROR: 500</h2>
<p>Problem accessing /services/assets/stass/_read. Reason:
<pre>    javax.servlet.ServletException: java.lang.NullPointerException</pre></p>
<hr /><a href="http://eclipse.org/jetty">Powered by Jetty:// 9.3.9.v20160517</a><hr/>
</body>
</html>
```

I recommend to maybe hide this exception and suggest to check e.g. the MAC.

Kura 3.1.0 RC b41 on RPi3

--- a/kura/org.eclipse.kura.driver.ble.sensortag.provider/src/main/java/org/eclipse/kura/internal/driver/ble/sensortag/SensorTagDriver.java
+++ b/kura/org.eclipse.kura.driver.ble.sensortag.provider/src/main/java/org/eclipse/kura/internal/driver/ble/sensortag/SensorTagDriver.java
@@ -255,10 +255,10 @@ public final class SensorTagDriver implements Driver, ConfigurableComponent {
     }
 
     private void runReadRequest(SensorTagRequestInfo requestInfo) {
-        TiSensorTag sensorTag = getSensorTag(requestInfo);
 
         ChannelRecord record = requestInfo.channelRecord;
         try {
+            TiSensorTag sensorTag = getSensorTag(requestInfo);
             Object readResult = getReadResult(requestInfo.sensorName, sensorTag);
             final Optional<TypedValue<?>> typedValue = getTypedValue(requestInfo.dataType, readResult);
             if (!typedValue.isPresent()) {
@@ -312,7 +312,7 @@ public final class SensorTagDriver implements Driver, ConfigurableComponent {
         }
     }
 
-    private TiSensorTag getSensorTag(SensorTagRequestInfo requestInfo) {
+    private TiSensorTag getSensorTag(SensorTagRequestInfo requestInfo) throws KuraBluetoothIOException {
         if (!this.tiSensorTagMap.containsKey(requestInfo.sensorTagAddress)) {
             Future<BluetoothLeDevice> future = this.bluetoothLeAdapter.findDeviceByAddress(TIMEOUT,
                     requestInfo.sensorTagAddress);
@@ -327,6 +327,9 @@ public final class SensorTagDriver implements Driver, ConfigurableComponent {
             }
         }
         TiSensorTag sensorTag = this.tiSensorTagMap.get(requestInfo.sensorTagAddress);
+        if (sensorTag == null) {
+            throw new KuraBluetoothIOException("Resource unavailable");
+        }
         if (!sensorTag.isConnected()) {
             connect(sensorTag);
         }
@@ -352,10 +355,18 @@ public final class SensorTagDriver implements Driver, ConfigurableComponent {
     }
 
     private void runWriteRequest(SensorTagRequestInfo requestInfo) {
-        TiSensorTag sensorTag = getSensorTag(requestInfo);
-
         ChannelRecord record = requestInfo.channelRecord;
         record.setTimestamp(System.currentTimeMillis());
+
+        TiSensorTag sensorTag;
+        try {
+            sensorTag = getSensorTag(requestInfo);
+        } catch (KuraBluetoothIOException e) {
+            record.setChannelStatus(new ChannelStatus(FAILURE, message.writeFailed(), null));
+            logger.error("IO Exception encountered while connecting.");
+            return;
+        }
+
         final TypedValue<?> value = record.getValue();
         if (!value.getType().equals(DataType.BOOLEAN)) {
             record.setChannelStatus(new ChannelStatus(FAILURE, message.writeFailed(), null));

